hls-test-utils/src/Test/Hls.hs

```diff
@@ -390,7 +622,7 @@ runSessionWithServer' ::
   FilePath ->
   Session a ->
   IO a
-runSessionWithServer' plugins conf sconf caps root s = withLock lock $ keepCurrentDirectory $ do
+runSessionWithServer' plugins conf sconf caps root s =  withLock lock $ keepCurrentDirectory $ do
```

Comment:
This is a UX thing, our tests seem to always run in parallel, but we use the lock to avoid spawning too many HLS sessions. When your first tasty test uses `runSessionWithServerInTmpDir'` and another uses `runSessionWithServer'`, then `runSessionWithServerInTmpDir'` needs to perform slightly more work than the other test case. Since all tests run in parallel, the test `runSessionWithServer'` reaches the lock before the first test and, thus, acquires the lock first. The first test will not run, until the other finished, and tasty will only print results until the first test has finished execution.

So, if you are really unlucky, you don't see test results until the whole test suite finished, and the execution times are also completely skewed.

Reply:
TBH, all these locks are annoying, I'd rather not have them at all. But refactoring that was outside my time budget for now. I think it is still an improvement.

Reply:
Yeah, fair enough. I wonder if we can't just use tasty's `NumThreads` option and set it to 1 on test trees that have `runSession` in them :thinking:

Reply:
I actually thought that was already the case.